src/Bridges/Constraint/geomean_to_exp.jl

```diff
@@ -0,0 +1,125 @@
+    w_func = [MOIU.eachscalar(MOI.get(model, MOI.ConstraintFunction(), exp_index))[3] for exp_index in bridge.exp_indices]
+    return MOIU.convert_approx(I, MOIU.operate(vcat, T, u_func, MOIU.vectorize(w_func)))
+end
+MOI.get(model::MOI.ModelLike, ::MOI.ConstraintSet, bridge::GeoMeantoExpBridge) = MOI.GeometricMeanCone(1 + length(bridge.z))
+MOI.supports(::MOI.ModelLike, ::Union{MOI.ConstraintPrimalStart, MOI.ConstraintDualStart},
+    ::Type{<:GeoMeantoExpBridge}) = true
+function MOI.get(model::MOI.ModelLike, attr::Union{MOI.ConstraintPrimal, MOI.ConstraintPrimalStart}, bridge::GeoMeantoExpBridge)
+    le_y_primal = MOI.get(model, attr, bridge.le_y_index)
+    exp_primals = MOI.get.(model, attr, bridge.exp_indices)
+    u_primal = le_y_primal + sum(exp_i_primal[2] for exp_i_primal in exp_primals) / length(exp_primals)
+    w_primal = [exp_i_primal[3] for exp_i_primal in exp_primals]
+    return vcat(u_primal, w_primal)
+end
+function MOI.set(model::MOI.ModelLike, attr::MOI.ConstraintPrimalStart, bridge::GeoMeantoExpBridge, value)
+    u_primal = value[1]
+    y_primal = max(zero(u_primal), u_primal)
+    MOI.set(model, MOI.VariablePrimalStart(), bridge.y, y_primal)
+    if iszero(y_primal)
+        z_primal = zeros(u_primal, length(bridge.z))
+    else
+        z_primal = [y_primal * log(value[i] / y_primal) for i in 2:length(value)]
+    end
+    MOI.set(model, MOI.VariablePrimalStart(), bridge.z, z_primal)
+    MOI.set(model, attr, bridge.le_y_index, u_primal - y_primal)
+    MOI.set(model, attr, bridge.ge_z_index, zero(u_primal))
+    for i in eachindex(bridge.z)
+        MOI.set(model, attr, bridge.exp_indices[i], vcat(z_primal[i], y_primal, value[1 + i]))
+    end
+    return
+end
+# (u, w) in GeoMean <=> exists y, z such that Au + Bw + Cy + Dz in (<=, >=, (exp)_i)
+# so GeoMean* = [A'; B'] (<=, >=, exp1, ..., expn)*
+# and 0 = [C'; D'] (<=, >=, exp1, ..., expn)*
+# where
+# A = [1, 0, (0, 0, 0)_i]
+# B = [0, 0, (0, 0, I)]
+# C = [-1, 0, (0, 1, 0)]
+# D = [0, 1, (I, 0, 0)]
+# so given dual q = (a, b, (c, d, e)_i), we get
+# u = A' q = a => a = u
+# w_i = B' q = e_i => e_i = w_i
+# 0 = C' q = -a + sum(d) => a = sum(d) = u
+# 0 = D' q = b + c_i => c_i = -b
+# Given a is dual on u - y <= 0, b is dual on sum(z) >= 0, and (c_i, d_i, e_i) is dual on ExponentialCone constraint i,
+# dual on (u, w) in GeometricMeanCone is (a, e).
+function MOI.get(model::MOI.ModelLike, attr::Union{MOI.ConstraintDual, MOI.ConstraintDualStart}, bridge::GeoMeantoExpBridge)
+    u_dual = MOI.get(model, attr, bridge.le_y_index)
+    w_dual = [MOI.get(model, attr, exp_index_i)[3] for exp_index_i in bridge.exp_indices]
+    return vcat(u_dual, w_dual)
+end
+# Given constraint dual start of (u, w), constraint dual on LessThan constraint is a = u,
+# on GreaterThan constraint is b, and on exponential cone constraint i is (c_i = -b, d_i, e_i = w_i).
+# We must have sum(d) = a = u. We let b = -c_i = -u/n and d_i = -b (log(w_i/b) + 1).
```

Comment:
Why is c_i equal to u / n?

Reply:
It was a choice (hence the "We let ...")

Reply:
I chose it because it allowed me to make that d_i value work. My math was sketchy though - it's worth double checking.

Reply:
You still have the same issue as in https://github.com/JuliaOpt/MathOptInterface.jl/pull/1017 that for the duals to be correct, you need to have `sum(d_i) = a`. With your choice of `d_i`, it's not necessarily the case.

Reply:
`sum(d) = sum_i -b (log(w_i/b) + 1) = -b (n + sum_i log(w_i/b)) = -b (n + log(prod(w_i/b))) = -(-u/n) (n + log(prod(w_i * -n/u))) = u + u/n * log(prod(w_i) * prod(-n/u))` which equals `u` if `prod(w_i) = prod(-u/n)` i.e. `-u = n * geomean(w)` i.e. if the dual geomean cone inequality holds at equality. but it's true that I can't assume that from `(u, w)`. so instead what if i let `b = -c_i = geomean(w)`. then I think it works?

Reply:
and `(-b, -b (log(w_i/b) + 1), w_i)` is still feasible for the dual exp cone `-u exp(v/u) <= exp(1) w` because:
```
b exp(-b (log(w_i/b) + 1) / -b) <= exp(1) w_i
b exp(log(w_i/b) + 1) <= exp(1) w_i
b * w_i/b * exp(1) <= exp(1) w_i
w_i <= w_i
```

Reply:
OK try again:

Given constraint dual start of `(u, w)`, constraint dual on LessThan constraint is `a = u`, on GreaterThan constraint is `b`, and on exponential cone constraint i is `(c_i = -b, d_i, e_i = w_i)`. We must have `sum(d) = a = u`. We let `b = -c_i = -u/n` and `d_i = -b * (log(w_i/geomean(w)) + 1)`. Note by dual geomean cone inequality, `-u <= n * geomean(w)` i.e. `geomean(w) >= -u/n`. Hence the dual on exponential cone constraint i is `(-b = u/n, d_i = u/n * (log(w_i/geomean(w)) + 1), w_i)`, which is feasible for the DualExponentialCone `(x,y,z): -x exp(y/x) <= exp(1) z`, since we have
```
-u/n * exp(u/n * (log(w_i/geomean(w)) + 1) / (u/n)) 
= -u/n * w_i / geomean(w) * exp(1) 
<= geomean(w) * w_i / geomean(w) * exp(1) 
= exp(1) * w_i.
```

Reply:
I think that's it